src/firefly/js/visualize/ui/ExtractionDialog.jsx

```diff
@@ -448,6 +451,37 @@ function ExtractionPanelView({pointSize, setPointSize, afterRedraw, plotlyDivSty
                                  plotlyData, canCreateExtractionTable,
                                  plotlyLayout, startUpHelp, callKeepExtraction,
                                  bottomUI, combineOp, setCombineOp, hasFloatingData}) {
+
+    const extractAddedChartInfo = (action) => {
+        const chartId = action.payload.chartId;
+        const tbl_id =   action.payload.data?.[0].tbl_id;
+        return {chartId, tbl_id};
+    };
```

Comment:
This function does not need to be a closure. Move it out of the component.
Also you can simply it.
```js
const extractAddedChartInfo = ({payload}) => ( {chartId: payload.chartId, tbl_id: payload.data?.[0].tbl_id});
```